src/network/networkd-setlink.c

```diff
@@ -87,6 +95,35 @@ static int link_configure(
                 return log_link_debug_errno(link, r, "Could not allocate RTM_SETLINK message: %m");
 
         switch (mode) {
+        case SET_LINK_FLAGS: {
+                unsigned ifi_change = 0, ifi_flags = 0;
+
+                if (link->network->arp >= 0) {
+                        SET_FLAG(ifi_change, IFF_NOARP, true);
```

Comment:
with a constant 3rd arg, i'd ust use `ifi_change |= IFF_NOARP` here... i mean, SET_FLAG() is really nice if the third arg is a variable, since it shortens stuff quite a bit here, but it's needlessly opaque if we just want to unconditionally OR something into a flags field...